### PR TITLE
Misc Changes / Fixes

### PR DESCRIPTION
**Pull request by artdude543**
Changed a lot of the code and formatting to make things look better in the code / UI wise. You can see more information in the commits to see what each change does.

All the changes have been tested / tweaked before creating this PR :)

### CODE REVIEW

**Comment by JamyDev** [comment]
Wow that's one helluva diff. Went through it all, looks good!

File: src/api/docs/channel/types.js [hunk 1]
```diff
@@ -21,12 +20,15 @@ export default class ChannelTypesMethod extends PaginatedMethod {
   parameters() {
     return super.parameters().concat([
       new SlugParameter(),
-      new OrderParameter(["online", "featured", "partnered", "name",
-                          "viewers_total", "followers", "subscribers"]),
-      new OnlyParameter(),
+      new OrderParameter(["online", "featured", "partnered", "name", "viewers_total", "followers", "subscribers"]),
```
Comment (connor4312) [inline src/api/docs/channel/types.js]: It's preferable to have things broken up onto multiple lines (usually wrapped within 79 characters)

File: src/components/api_reference_page.jsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ export default class APIReferencePage extends React.Component {
   }
 
   render() {
+    let defaultName = "achievements";
```
Comment (connor4312) [inline src/components/api_reference_page.jsx]: unused?

File: src/api/orderParameter.js [hunk 3]
```diff
@@ -14,9 +14,8 @@ export default class OrderParameter extends Parameter {
     return (
       <div>
         <p>
-          Specifies the order in which items should be displayed in the format
-           <code>attribute:order</code>. Order should be one of <code>asc</code>
-           or <code>desc</code>. Attribute may be one of:
+          Specifies the order in which items should be displayed in the form <code>attribute:order</code>.
+          Order should be one of <code>asc</code> or <code>desc</code>. Attribute may be one of:
```
Comment (connor4312) [inline src/api/orderParameter.js]: Again, wrapping stuff

File: src/components/api_reference_page.jsx (hunk 2 above)
Comment (JamyDev) [inline src/components/api_reference_page.jsx]: I removed this in an earlier commit but apparently it doesn't work without...

File: src/components/method/example_group.js [hunk 4]
```diff
@@ -14,7 +14,11 @@ export default class ExampleGroup extends React.Component {
     return (
       <div className="examples">
         <ul className="examples-list">{
-          this.examples().map((ex) => <li>{ ex }</li>)
+          this.examples().map((ex) => {
+            let httpCode = ex.props.example.httpCode();
+
+            return <li key={httpCode + ":" + Math.random()}>{ ex }</li>;
```
Comment (connor4312) [inline src/components/method/example_group.js]: `Array.prototype.map` provides the index of the element as the second argument to the iterating function. Use that.

File: src/components/method/method_group.js [hunk 5]
```diff
@@ -15,7 +15,11 @@ export default class MethodGroup extends React.Component {
     return (
       <div className="method-group" id={this.props.name}>
         <ul className="methods">{
-          this.methods().map((m) => <li>{ m }</li>)
+          this.methods().map((m) => {
+            let href = `#${m.props.endpoint.httpMethod().toLowerCase()}:${m.props.endpoint.uri().replace(/:|\//g, "")}`;
+
+            return <li key={href}>{ m }</li>;
```
Comment (connor4312) [inline src/components/method/method_group.js]: Again, you can just use the index as the key here.

File: src/components/method/parameter_group.js [hunk 6]
```diff
@@ -12,12 +13,18 @@ export default class ParameterGroup extends React.Component {
...
-        }</ul>
-      </div>
+      <If test={this.parameters().length > 0}>
+        <div className="parameters">
+          <span className="parameters-header">Parameters</span>
+          <ul className="parameters-list">{
+            this.parameters().map((parameter) => {
+              let name = parameter.props.parameter.name();
+
+              return <li key={name}>{ parameter }</li>;
```
Comment (connor4312) [inline src/components/method/parameter_group.js]: keyyyy

**Comment by connor4312** [comment]
Adding in eslint would be good to actually enforce the stylistic changes contained here. Main thing is line length. Most of our projects have a hard max of 120 with a recommended wrap at column 79 (following in PEP8's footsteps).

File: src/components/api_reference_page.jsx (hunk 2 above)
Comment (artdude543) [inline src/components/api_reference_page.jsx]: This is needed for some bizarre reason. I have no idea why the logic is what is but :P

**Comment by artdude543** [comment]
Okay did the changes mentioned and added a 404 page.

**Comment by ttaylorr** [comment]
Thanks for this! I agree with your suggestion, let's get ESLint in here to make sure that these standards are maintained.

**Comment by JamyDev** [comment]
Will add eslint in a separate PR.

**Comment by ttaylorr** [comment]
Should be pretty simple to integrate that in this one. @artdude543: what are your thoughts?

**Comment by artdude543** [comment]
Umm, if you send me the lint file of what you want I can go through it and update the code to meet it. Otherwise I can do that in another PR after this one is merged.

**Comment by ttaylorr** [comment]
How about we work backwards from the code style in this PR to an .eslintrc file? Giving it more thought, we could totally add this in a separate PR instead :smile:

**Comment by artdude543** [comment]
Sounds like a plan :) I don't mind helping out with the site so assign me jobs and I'll get around to them after work / other projects. To help take some load of you guys working on the main site ;)

**Comment by connor4312** [comment]
I usually use Airbnb's eslint rules as a starting place. [repo](https://github.com/airbnb/javascript)

**Comment by JamyDev** [comment]
Yes linting will go in a separate PR since there's probably a lot of things to change after adding it.

**Pull request merged**